Tests/Runner/BranchedWorkspaceTest.php

```diff
@@ -0,0 +1,281 @@
+    {
+        try {
+            $componentsApi->deleteConfiguration($configuration->getComponentId(), $configuration->getConfigurationId());
+        } catch (ClientException $e) {
+            if ($e->getCode() !== 404) {
+                throw $e;
+            }
+        }
+
+        return $this->createConfiguration($componentsApi, $configuration);
+    }
+
+    private function createConfiguration(Components $componentsApi, Configuration $configuration)
+    {
+        return $componentsApi->addConfiguration($configuration)['id'];
+    }
+
+    private function cleanupAndCreateBranch(DevBranches $branchesApi, $branchName)
+    {
+        $branchList = $branchesApi->listBranches();
+        foreach ($branchList as $branch) {
+            if ($branch['name'] === $branchName) {
+                $branchesApi->deleteBranch($branch['id']);
+                break;
+            }
+        }
+
+        return $branchesApi->createBranch($branchName)['id'];
+    }
+
+    private function loadDataIntoTable(Client $storageApi, $bucketId, $tableName, array $data)
+    {
+        $temp = new Temp();
+        $temp->initRunFolder();
+
+        $csv = new CsvFile($temp->getTmpFolder() . '/upload.csv');
+        foreach ($data as $row) {
+            $csv->writeRow($row);
+        }
+
+        $storageApi->createTableAsync($bucketId, $tableName, $csv);
+    }
+
+    private function runCopyJob(ClientWrapper $storageApiWrapper, $configId, $inBucketId, $outBucketId, $tableName)
+    {
+        $configData = [
+            'storage' => [
+                'input' => [
+                    'tables' => [
+                        [
+                            'source' => $inBucketId.'.'.$tableName,
+                            'destination' => 'local-table',
+                        ],
+                    ],
+                ],
+                'output' => [
+                    'tables' => [
+                        [
+                            'source' => 'local-table',
+                            'destination' => $outBucketId.'.'.$tableName,
+                        ],
+                    ],
+                ],
+            ],
+            'parameters' => [
+                'operation' => 'copy',
+            ],
+        ];
+
+        $componentData = [
+            'id' => 'keboola.runner-workspace-test',
+            'data' => [
+                'definition' => [
+                    'type' => 'aws-ecr',
+                    'uri' => '061240556736.dkr.ecr.us-east-1.amazonaws.com/keboola.runner-workspace-test',
+                    'tag' => 'latest',
+                ],
+                'staging_storage' => [
+                    'input' => 'workspace-snowflake',
+                    'output' => 'workspace-snowflake',
+                ],
+            ],
+        ];
+
+        $this->runJob($storageApiWrapper, $configData, $componentData, $configId);
```

Comment:
imho programuješ v testech - ty vnořený volání .
`cleanupAndCreateBucket`, `loadDataIntoTable`, `cleanupAndCreateBranch` - je super
ale připadá mi, že `runCopyJob` by mělo být spíš `getCopyJobParameters` a `runJob` - na vysvětlení "proč" bych nad tím ale asi potřeboval ještě zapřemýšlet

Reply:
Tohle trochu souvisi s tim, ze tam nejsou asserty. Premyslel jsem nad tim tak, ze smyslem toho testu je popsat a otestovat to chovani (kdyz zalozim config pred/po/v branchi) a aby to z toho bylo co nejsnadneji videt. Proto jsem vsechnu "omacku" schoval do metod.

To ze je tam `runCopyJob` a `runJob` je spis jen postupem, jakym jsme ty metody vytvarel. Nejdriv jsem si udelal generickou `runJob` (ktera je teoreticky reusnutelna napric testy) a pak kdyz jsme videl, ze vsechny test-cases pouzivaji uplne stejnou configuraci (configData/componentData) a vlastne tim padem to neni to, co testuji, tak jsem to vytahnul do metody jeste jednou.

Reply:
Rozdelil jsem na 2 metody dle domluvy